modules/coremark/core_portme.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "coremark.h"
+
+#if CONFIG_COREMARK_RUN_TYPE_VALIDATION
+volatile ee_s32 seed1_volatile = 0x3415;
+volatile ee_s32 seed2_volatile = 0x3415;
+volatile ee_s32 seed3_volatile = 0x66;
+#endif
+
+#if CONFIG_COREMARK_RUN_TYPE_PERFORMANCE
+volatile ee_s32 seed1_volatile = 0x0;
+volatile ee_s32 seed2_volatile = 0x0;
+volatile ee_s32 seed3_volatile = 0x66;
+#endif
+
+#if CONFIG_COREMARK_RUN_TYPE_PROFILE
+volatile ee_s32 seed1_volatile = 0x8;
+volatile ee_s32 seed2_volatile = 0x8;
+volatile ee_s32 seed3_volatile = 0x8;
+#endif
+
+volatile ee_s32 seed4_volatile = CONFIG_COREMARK_ITERATIONS;
+volatile ee_s32 seed5_volatile = 0;
+
+ee_u32 default_num_contexts = CONFIG_COREMARK_THREADS_NUMBER;
+
+static CORE_TICKS start_time_val;
+static CORE_TICKS stop_time_val;
+
+void start_time(void)
+{
+	start_time_val = k_cycle_get_32();
+}
+
+void stop_time(void)
+{
+	stop_time_val = k_cycle_get_32();
+}
+
+CORE_TICKS get_time(void)
+{
+	return (stop_time_val - start_time_val);
+}
+
+secs_ret time_in_secs(CORE_TICKS ticks)
+{
+	secs_ret retval = (secs_ret)(k_cyc_to_ns_floor64(ticks) / NSEC_PER_USEC) / USEC_PER_SEC;
+	return retval;
+}
+
+void portable_init(core_portable *p, int *argc, char *argv[])
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	if (sizeof(ee_ptr_int) != sizeof(ee_u8 *)) {
+		ee_printf(
+			"ERROR! Please define ee_ptr_int to a type that holds a "
+			"pointer!\n");
+	}
+
+	if (sizeof(ee_u32) != 4) {
+		ee_printf("ERROR! Please define ee_u32 to a 32b unsigned type!\n");
+	}
+
+	p->portable_id = 1;
+}
+
+void portable_fini(core_portable *p)
+{
+	p->portable_id = 0;
+}
+
+COREMARK_THREAD_STACK_INSTANCE_DEFINE(pstack, CONFIG_COREMARK_THREADS_NUMBER);
+
+static struct k_thread coremark_thread_data[CONFIG_COREMARK_THREADS_NUMBER];
+
+static int active_threads;
```

Comment:
We typically keep the static variables at the top of the file. Could we move them there?

Also we can use prefixes for static variables to highlight that they belong to the same category / are closely relate. In this context, I would go for `thread_`

```suggestion
COREMARK_THREAD_STACK_INSTANCE_DEFINE(thread_stacks, CONFIG_COREMARK_THREADS_NUMBER);

static struct k_thread thread_descriptors[CONFIG_COREMARK_THREADS_NUMBER];

static int thread_cnt;
```